test/databases/transaction_database.cpp

```diff
@@ -38,51 +38,502 @@ struct transaction_database_directory_setup_fixture
+
+BOOST_AUTO_TEST_CASE(transaction_database__store__single_unconfirmed_without_a_block__success)
+{
+    transaction tx1;
+    data_chunk wire_tx1;
+    BOOST_REQUIRE(decode_base16(wire_tx1, TRANSACTION1));
+    BOOST_REQUIRE(tx1.from_data(wire_tx1));
+    
+    const auto path = DIRECTORY "/tx_table";
+    test::create(path);
+    transaction_database db(path, 1000, 50, 0);
+    BOOST_REQUIRE(db.create());
+
+    const auto hash1 = tx1.hash();
+    BOOST_REQUIRE(!db.get(hash1));
+    
+    // Setup end
+    
+    db.store(tx1, 1);
+
+    const auto result1 = db.get(hash1);
+    BOOST_REQUIRE(result1);
+    BOOST_REQUIRE(result1.transaction().hash() == hash1);
+    
+    // Verify computed hash and get via link
+    const auto fetched = db.get(result1.link());
+    BOOST_REQUIRE(fetched);
+    BOOST_REQUIRE(fetched.transaction().hash() == hash1);
+
+}
+
+BOOST_AUTO_TEST_CASE(transaction_database__store__list_unconfirmed_without_a_block__success)
+{
+    transaction tx1;
+    data_chunk wire_tx1;
+    BOOST_REQUIRE(decode_base16(wire_tx1, TRANSACTION1));
+    BOOST_REQUIRE(tx1.from_data(wire_tx1));
+
+    transaction tx2;
+    data_chunk wire_tx2;
+    BOOST_REQUIRE(decode_base16(wire_tx2, TRANSACTION2));
+    BOOST_REQUIRE(tx2.from_data(wire_tx2));
+
+    const auto path = DIRECTORY "/tx_table";
+    test::create(path);
+    transaction_database db(path, 1000, 50, 0);
+    BOOST_REQUIRE(db.create());
+
+    const auto hash1 = tx1.hash();
+    BOOST_REQUIRE(!db.get(hash1));
+
+    const auto hash2 = tx2.hash();
+    BOOST_REQUIRE(!db.get(hash2));
+
+    // Setup end
+    
+    db.store({tx1, tx2});
+
+    const auto result1 = db.get(hash1);
+    BOOST_REQUIRE(result1);
+    BOOST_REQUIRE(result1.transaction().hash() == hash1);
+
+    const auto result2 = db.get(hash2);
+    BOOST_REQUIRE(result2);
+    BOOST_REQUIRE(result2.transaction().hash() == hash2);
+
+    // Verify computed hash and get via link
+    const auto result3 = db.get(result2.link());
+    BOOST_REQUIRE(result3);
+    BOOST_REQUIRE(result3.transaction().hash() == hash2);
+
+}
+
+
+// Candidate/Uncandidate
+// ----------------------------------------------------------------------------
+
+BOOST_AUTO_TEST_CASE(transaction_database__candidate__with_no_input_in_db__returns_false)
+{
+    transaction tx1;
+    data_chunk wire_tx1;
+    BOOST_REQUIRE(decode_base16(wire_tx1, TRANSACTION1));
+    BOOST_REQUIRE(tx1.from_data(wire_tx1));
+
+    const auto path = DIRECTORY "/tx_table";
+    test::create(path);
+    transaction_database db(path, 1000, 50, 0);
+    BOOST_REQUIRE(db.create());
+
+    const auto hash1 = tx1.hash();
+    db.store(tx1, 1);
+
+    const auto result1 = db.get(hash1);
+    BOOST_REQUIRE(result1);
+    BOOST_REQUIRE(result1.transaction().hash() == hash1);
+    
+    // Setup end    
+
+    const bool result = db.candidate(db.get(hash1).link());
+    BOOST_REQUIRE(!result);
+
+}
+
+BOOST_AUTO_TEST_CASE(transaction_database__candidate__with_input_in_db__returns_true)
```

Comment:
This isn't a return value, it's the value of `candidate_spend` being tested, so name:
```cpp
BOOST_AUTO_TEST_CASE(transaction_database__candidate__with_input_in_db__candidate_spend_true)
```

Reply:
Fixed, in other tests too.